app/lib/services/import.dart

```diff
@@ -413,50 +454,75 @@ class ImportService {
-          final scale = 1 / quality;
-          final height = page.height;
-          final width = page.width;
-          final dataPath = Uri.dataFromBytes(png).toString();
-          final element = ImageElement(
-              height: height.toDouble(),
-              width: width.toDouble(),
-              source: dataPath,
-              constraints:
-                  ElementConstraints.scaled(scaleX: scale, scaleY: scale),
-              position: Point(firstPos.dx, y));
-          final area = Area(
-            height: height * scale,
-            width: width * scale,
-            position: Point(firstPos.dx, y),
-            name: localizations.pageIndex(areas.length + 1),
-          );
-          if (spreadToPages) {
-            documentPages.add(DocumentPage(
-              content: [element],
-              areas: [if (createAreas) area],
-            ));
-          } else {
-            selectedElements.add(element);
-            if (createAreas) {
-              areas.add(area);
+          try {
+            await Future.delayed(const Duration(milliseconds: 100));
+            dialog?.setProgress(current / pages.length);
+            current++;
+            final png = await page.toPng();
+            final scale = 1 / quality;
+            final height = page.height;
+            final width = page.width;
+            final dataPath = Uri.dataFromBytes(png).toString();
+            final element = ImageElement(
+                height: height.toDouble(),
+                width: width.toDouble(),
+                source: dataPath,
+                constraints:
+                    ElementConstraints.scaled(scaleX: scale, scaleY: scale),
+                position: Point(firstPos.dx, y));
+            final area = Area(
+              height: height * scale,
+              width: width * scale,
+              position: Point(firstPos.dx, y),
+              name: localizations.pageIndex(areas.length + 1),
+            );
+            if (spreadToPages) {
+              documentPages.add(DocumentPage(
+                content: [element],
+                areas: [if (createAreas) area],
+              ));
+            } else {
+              selectedElements.add(element);
+              if (createAreas) {
+                areas.add(area);
+              }
             }
+            y += height * scale;
+          } catch (e) {
+            showDialog(
+              context: context,
+              builder: (context) =>
+                  UnknownImportConfirmationDialog(message: e.toString()),
+            );
           }
-          y += height * scale;
         }
+
         dialog?.close();
-        return _submit(
-          document,
-          elements: selectedElements,
-          pages: documentPages,
-          areas: createAreas ? areas : [],
-          choosePosition: position == null,
-        );
+
+        // if count image are > 50 choosePosition: position != null,
+        if (elements.length > 50) {
```

Comment:
I'm currently not sure what you do here?
choosePosition is true if elements.length > 50 and position != null?

Reply:
If the number of pages is greater than 50 I prevent the choice of the position of the PDF on the canvas

Reply:
Yes, The app crashes

Reply:
Then please use choosePosition: false